S3 PutObject errors with 'Data read has a different checksum than expected'

The latest release of Localstack is causing the AWS SDK to throw the following error:

```
software.amazon.awssdk.core.exception.SdkClientException: Unable to unmarshall response (Data read has a different checksum than expected.). Response Code: 200, Response Text: OK

	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:97)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.handleSuccessResponse(HandleResponseStage.java:100)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.handleResponse(HandleResponseStage.java:70)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:58)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:41)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:73)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:77)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:39)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage$RetryExecutor.doExecute(RetryableStage.java:113)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage$RetryExecutor.execute(RetryableStage.java:86)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:62)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:57)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.executeWithTimer(ApiCallTimeoutTrackingStage.java:80)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:60)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26)
	at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:240)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:96)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:120)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:73)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:44)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:55)
	at software.amazon.awssdk.services.s3.DefaultS3Client.putObject(DefaultS3Client.java:3056)
	<!-- SNIP -->
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:628)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:117)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:184)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:180)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:127)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: software.amazon.awssdk.core.exception.SdkClientException: Data read has a different checksum than expected.
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:97)
	at software.amazon.awssdk.core.exception.SdkClientException.create(SdkClientException.java:39)
	at software.amazon.awssdk.services.s3.checksums.ChecksumsEnabledValidator.validatePutObjectChecksum(ChecksumsEnabledValidator.java:134)
	at software.amazon.awssdk.services.s3.internal.handlers.SyncChecksumValidationInterceptor.afterUnmarshalling(SyncChecksumValidationInterceptor.java:90)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.lambda$afterUnmarshalling$9(ExecutionInterceptorChain.java:152)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.reverseForEach(ExecutionInterceptorChain.java:210)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.afterUnmarshalling(ExecutionInterceptorChain.java:152)
	at software.amazon.awssdk.core.client.handler.BaseClientHandler.runAfterUnmarshallingInterceptors(BaseClientHandler.java:138)
	at software.amazon.awssdk.core.client.handler.BaseClientHandler.lambda$interceptorCalling$2(BaseClientHandler.java:151)
	at software.amazon.awssdk.core.client.handler.AttachHttpMetadataResponseHandler.handle(AttachHttpMetadataResponseHandler.java:40)
	at software.amazon.awssdk.core.client.handler.AttachHttpMetadataResponseHandler.handle(AttachHttpMetadataResponseHandler.java:28)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.handleSuccessResponse(HandleResponseStage.java:89)
	... 85 more
```

The code causing this exception is:

```
s3Client.putObject(PutObjectRequest.builder()
    .bucket(bucketName)
    .key(key)
    .build(),
    RequestBody.empty());
```

(Note that this is using [aws-sdk-java-v2](https://github.com/aws/aws-sdk-java-v2))

I have also tested with a `RequestBody` made from a string, e.g. `            RequestBody.fromString("My text")` but that also caused the same exception.

Version 0.10.3 does not throw this error. I have pointed my code at the real S3 endpoint and verified that it does not throw the error. I tried to replicate the behaviour in Localstack with the AWS CLI but I was not able to, however I'm not sure how equivalent the command I used is to the code above:

```
aws --endpoint-url http://localhost:4572 s3api put-object --bucket local-content --key empty-file
```

The only commit I can see between version 0.10.3 and the latest release that may be the cause is for PR #1608 but I can't figure out how it could have caused this issue.

Is this a bug in Localstack or is there a problem with my code?

--- a/Makefile
+++ b/Makefile
@@ -97,7 +97,7 @@ test:              ## Run automated tests
 		($(VENV_RUN); DEBUG=$(DEBUG) PYTHONPATH=`pwd` nosetests --with-timer --with-coverage --logging-level=WARNING --nocapture --no-skip --exe --cover-erase --cover-tests --cover-inclusive --cover-package=localstack --with-xunit --exclude='$(VENV_DIR).*' --ignore-files='lambda_python3.py' $(TEST_PATH))
 
 test-java:         ## Run tests for Java/JUnit compatibility
-	cd localstack/ext/java; USE_SSL=1 mvn -q test
+	cd localstack/ext/java; USE_SSL=1 SERVICES=serverless,kinesis,sns,sqs mvn -q test
 
 prepare-java-tests-if-changed:
 	@(! (git log -n 1 --no-merges --raw | grep localstack/ext/java/)) || (\

--- a/localstack/ext/java/pom.xml
+++ b/localstack/ext/java/pom.xml
@@ -31,6 +31,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
+        <aws.sdk.version>1.11.642</aws.sdk.version>
     </properties>
 
     <dependencies>
@@ -43,12 +44,6 @@
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-api</artifactId>
-            <version>5.1.0</version>
-            <optional>true</optional>
-        </dependency>
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-engine</artifactId>
             <version>5.1.0</version>
             <optional>true</optional>
         </dependency>
@@ -82,9 +77,16 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+            <version>1.2</version>
+        </dependency>
+
+        <!-- AWS SDKs -->
+        <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk</artifactId>
-            <version>1.11.615</version>
+            <version>${aws.sdk.version}</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
@@ -95,27 +97,22 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-lambda-java-events</artifactId>
-            <version>2.2.5</version>
-        </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
-            <version>1.2</version>
+            <version>2.2.7</version>
         </dependency>
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-core</artifactId>
-            <version>1.11.505</version>
+            <version>${aws.sdk.version}</version>
         </dependency>
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-lambda</artifactId>
-            <version>1.11.505</version>
+            <version>${aws.sdk.version}</version>
         </dependency>
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-sqs</artifactId>
-            <version>1.11.505</version>
+            <version>${aws.sdk.version}</version>
         </dependency>
 
 

--- a/localstack/ext/java/src/main/java/cloud/localstack/Localstack.java
+++ b/localstack/ext/java/src/main/java/cloud/localstack/Localstack.java
@@ -79,9 +79,7 @@ public class Localstack {
         }
     }
 
-    private Localstack() {
-
-    }
+    private Localstack() { }
 
     public static boolean isDevEnvironment() {
         return DEV_ENVIRONMENT;

--- a/localstack/ext/java/src/main/java/cloud/localstack/TestUtils.java
+++ b/localstack/ext/java/src/main/java/cloud/localstack/TestUtils.java
@@ -46,6 +46,10 @@ public class TestUtils {
     public static final String TEST_SECRET_KEY = "test";
     public static final AWSCredentials TEST_CREDENTIALS = new BasicAWSCredentials(TEST_ACCESS_KEY, TEST_SECRET_KEY);
 
+    private static final String[] EXCLUDED_DIRECTORIES = {
+      ".github", ".git", ".idea", ".venv", "target", "node_modules"
+    };
+
     public static void setEnv(String key, String value) {
         Map<String, String> newEnv = new HashMap<String, String>(System.getenv());
         newEnv.put(key, value);
@@ -230,22 +234,24 @@ public class TestUtils {
 
     public static void copyFolder(Path src, Path dest) throws IOException {
         try(Stream<Path> stream = Files.walk(src)) {
-            stream.forEach(source -> copy(source, dest.resolve(src.relativize(source))));
+            stream.forEach(source -> {
+                boolean isExcluded = Arrays.stream(EXCLUDED_DIRECTORIES)
+                    .anyMatch( excluded -> source.toAbsolutePath().toString().contains(excluded));
+                if (!isExcluded) {
+                    copy(source, dest.resolve(src.relativize(source)));
+                }
+            });
         }
     }
-
-    private static String[] excludedDirectories = {".github", ".git", ".idea", ".venv", "target", "node_modules"};
 
     public static void copy(Path source, Path dest) {
         try {
             CopyOption[] options = new CopyOption[] {StandardCopyOption.COPY_ATTRIBUTES, StandardCopyOption.REPLACE_EXISTING};
+            if(Files.isDirectory(dest)) {
+                // continue without copying
+                return;
+            }
             if (Files.exists(dest)) {
-                if(Files.isDirectory(dest)
-                        || Arrays.stream(excludedDirectories)
-                            .anyMatch( excluded -> source.toAbsolutePath().toString().contains(excluded))) {
-                    // continue without copying
-                    return;
-                }
                 try(FileChannel sourceFile = FileChannel.open(source)) {
                     try (FileChannel destFile = FileChannel.open(dest)) {
                         if (!Files.getLastModifiedTime(source).equals(Files.getLastModifiedTime(dest))

--- a/localstack/ext/java/src/main/java/cloud/localstack/docker/Container.java
+++ b/localstack/ext/java/src/main/java/cloud/localstack/docker/Container.java
@@ -29,6 +29,8 @@ public class Container {
     private static final long POLL_INTERVAL = 1000;
     private static final int NUM_LOG_LINES = 10;
 
+    private static final String ENV_DEBUG = "DEBUG";
+    private static final String ENV_DEBUG_DEFAULT = "1";
     public static final String LOCALSTACK_EXTERNAL_HOSTNAME = "HOSTNAME_EXTERNAL";
 
 
@@ -56,6 +58,7 @@ public class Container {
         String containerId = new RunCommand(LOCALSTACK_NAME, imageTag)
                 .withExposedPorts(LOCALSTACK_PORTS, randomizePorts)
                 .withEnvironmentVariable(LOCALSTACK_EXTERNAL_HOSTNAME, externalHostName)
+                .withEnvironmentVariable(ENV_DEBUG, ENV_DEBUG_DEFAULT)
                 .withEnvironmentVariables(environmentVariables)
                 .execute();
         LOG.info("Started container: " + containerId);

--- a/localstack/ext/java/src/main/java/cloud/localstack/docker/LocalstackDocker.java
+++ b/localstack/ext/java/src/main/java/cloud/localstack/docker/LocalstackDocker.java
@@ -26,7 +26,7 @@ public class LocalstackDocker {
     private static final Logger LOG = Logger.getLogger(LocalstackDocker.class.getName());
 
     private static final String PORT_CONFIG_FILENAME = "/opt/code/localstack/" +
-            ".venv/lib/python3.7/site-packages/localstack_client/config.py";
+            ".venv/lib/python3.6/site-packages/localstack_client/config.py";
 
     private static final Pattern READY_TOKEN = Pattern.compile("Ready\\.");
 
@@ -188,7 +188,7 @@ public class LocalstackDocker {
         }
 
         if (!serviceToPortMap.containsKey(serviceName)) {
-            throw new IllegalArgumentException("Unknown port mapping for service");
+            throw new IllegalArgumentException("Unknown port mapping for service: " + serviceName);
         }
 
         int internalPort = serviceToPortMap.get(serviceName);

--- a/localstack/ext/java/src/test/java/cloud/localstack/BasicFunctionalityTest.java
+++ b/localstack/ext/java/src/test/java/cloud/localstack/BasicFunctionalityTest.java
@@ -106,6 +106,7 @@ public class BasicFunctionalityTest {
         request.setRuntime(Runtime.Java8);
         request.setCode(LocalTestUtil.createFunctionCode(KinesisLambdaHandler.class));
         request.setHandler(KinesisLambdaHandler.class.getName());
+        request.setRole("r1");
         lambda.createFunction(request);
 
         // create stream
@@ -138,6 +139,7 @@ public class BasicFunctionalityTest {
         CreateFunctionRequest request = new CreateFunctionRequest();
         request.setFunctionName(functionName);
         request.setRuntime(Runtime.Java8);
+        request.setRole("r1");
         if(Localstack.useSSL()){
             request.setCode(LocalTestUtil.createFunctionCode(SQSLambdaHandlerSSL.class));
             request.setHandler(SQSLambdaHandlerSSL.class.getName());

--- a/localstack/ext/java/src/test/java/cloud/localstack/LocalTestUtil.java
+++ b/localstack/ext/java/src/test/java/cloud/localstack/LocalTestUtil.java
@@ -2,6 +2,7 @@ package cloud.localstack;
 
 import java.io.ByteArrayOutputStream;
 import java.io.File;
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.jar.JarEntry;
@@ -35,10 +36,15 @@ public class LocalTestUtil {
 		addClassToJar(clazz, jarStream);
 		addClassToJar(Record.class, jarStream);
 		addClassToJar(SQSEvent.class, jarStream);
+		// write MANIFEST into jar stream
+		JarEntry mfEntry = new JarEntry("META-INF/MANIFEST.MF");
+		jarStream.putNextEntry(mfEntry);
+		jarStream.closeEntry();
 		jarStream.close();
+
 		// write jar into zip stream
-		ZipEntry zipEntry = new ZipEntry("LambdaCode.jar");
-		zipStream.putNextEntry(zipEntry);
+		ZipEntry codeEntry = new ZipEntry("LambdaCode.jar");
+		zipStream.putNextEntry(codeEntry);
 		zipStream.write(jarOut.toByteArray());
 		zipStream.closeEntry();
 

--- a/localstack/services/awslambda/lambda_api.py
+++ b/localstack/services/awslambda/lambda_api.py
@@ -463,7 +463,7 @@ def get_java_handler(zip_file_content, handler, main_file):
             return result
         return execute, zip_file_content
     raise ClientError(error_response(
-        'Unable to extract Java Lambda handler - file is not a valid zip/jar files', 400, error_type='ValidationError'))
+        'Unable to extract Java Lambda handler - file is not a valid zip/jar file', 400, error_type='ValidationError'))
 
 
 def set_archive_code(code, lambda_name, zip_file_content=None):

--- a/localstack/services/s3/s3_listener.py
+++ b/localstack/services/s3/s3_listener.py
@@ -346,7 +346,7 @@ def strip_chunk_signatures(data):
     # <payload data ...>
     # 0;chunk-signature=927ab45acd82fc90a3c210ca7314d59fedc77ce0c914d79095f8cc9563cf2c70
 
-    data_new = re.sub(b'[0-9a-fA-F]+;chunk-signature=[0-9a-f]{64}(\r\n)', b'',
+    data_new = re.sub(b'(^|\r\n)[0-9a-fA-F]+;chunk-signature=[0-9a-f]{64}(\r\n)(\r\n$)?', b'',
         data, flags=re.MULTILINE | re.DOTALL)
 
     return data_new

--- a/tests/integration/test_s3.py
+++ b/tests/integration/test_s3.py
@@ -260,7 +260,7 @@ class S3ListenerTest (unittest.TestCase):
         headers = dict([[field.strip() for field in pair.strip().split(':', 1)]
             for pair in headers.strip().split('\n')])
         data = ('d;chunk-signature=af5e6c0a698b0192e9aa5d9083553d4d241d81f69ec62b184d05c509ad5166af\r\n' +
-            '%s0;chunk-signature=f2a50a8c0ad4d212b579c2489c6d122db88d8a0d0b987ea1f3e9d081074a5937\r\n') % body
+            '%s\r\n0;chunk-signature=f2a50a8c0ad4d212b579c2489c6d122db88d8a0d0b987ea1f3e9d081074a5937\r\n') % body
         # put object
         url = '%s/%s/%s' % (os.environ['TEST_S3_URL'], bucket_name, object_key)
         req = PutRequest(url, to_bytes(data), headers)

